base/dict.jl

```diff
@@ -187,12 +189,13 @@ function merge!(d::Associative, others::Associative...)
     end
     return d
 end
+keytype{K,V}(::Associative{K,V}) = K
+valtype{K,V}(::Associative{K,V}) = V
```

Comment:
Maybe, but one cannot hesitate enough to add new exports. We might instead want a better way to reflect on tuple types. See https://github.com/JuliaLang/julia/pull/10861#issuecomment-94050825 and the following comment for example.